buildSrc/src/main/kotlin/io/opentelemetry/gradle/OtelVersionClassPlugin.kt

```diff
@@ -0,0 +1,74 @@
+package io.opentelemetry.gradle
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.JavaPluginExtension
+import org.gradle.kotlin.dsl.the
+import java.io.File
+
+/**
+ * This gradle plugin will define a new task called generateOtelVersionClass.
+ * This task generates a Java source file that contains the project version
+ * as a string constant. The "compileJava" task is updated to depend on
+ * generateOtelVersionClass, and the project source set is updated to
+ * include the new file.
+ */
+class OtelVersionClassPlugin : Plugin<Project> {
+  override fun apply(project: Project) {
+    project.plugins.apply(JavaPlugin::class.java)
+
+    project.task("generateOtelVersionClass") {
+      doLast {
+        writeFile(project)
+      }
+    }
+    addTaskDependency(project)
+    addSourceDependency(project)
```

Comment:
#nit: would prefer to see the gradle task setup in one place, rather than having to step into these methods to see what's happening

Reply:
I thought the names gave pretty clear abstractions, but I can inline if that's preferred.